src/app/shell/workshop-details/side-menu/side-menu.component.ts

```diff
@@ -8,11 +12,17 @@ import { Workshop } from 'src/app/shared/models/workshop.model';
 })
 export class SideMenuComponent implements OnInit {
 
+  isDisplayed: boolean = false;
+  isRegistered: boolean = false;
+
   @Input() workshop: Workshop;
 
-  constructor() { }
+  constructor(private store: Store) { }
 
   ngOnInit(): void {
-  }
+    const user = this.store.selectSnapshot<User>(RegistrationState.user);
+    this.isRegistered = Boolean(user);
+    this.isDisplayed=Boolean((!user) || (user && user.role !== Role.provider))
```

Comment:
this.isRegistered = Boolean(user);
this.isDisplayed = (user?.role !== Role.provider);